projects/seeker/agents/gpt2_seeker_modules.py

```diff
@@ -0,0 +1,553 @@
+            if not search_results:
+                search_results = self._empty_docs(self.n_docs)  # type: ignore
+            elif len(search_results) < self.n_docs:  # type: ignore
+                remain_docs = self.n_docs - len(search_results)  # type: ignore
+                search_results.extend(self._empty_docs(remain_docs))  # type: ignore
+            docs_i = []
+            scors_i = []
+            # Change this debug later
+            logging.debug(
+                f'URLS:\n{self._display_urls(search_results)}'
+            )  # type: ignore
+            label_text = self.dict.vec2txt(
+                self.label_vec[batch_id, :-1]
+            )  # type: ignore
+            for i, doc in enumerate(search_results):
+                url = doc['url']
+                title = doc['title']
+                dcontent = doc['content']
+                assert type(dcontent) in (
+                    str,
+                    list,
+                ), f'Unrecognized retrieved doc: {dcontent}'
+                full_text = (
+                    dcontent if isinstance(dcontent, str) else '\n'.join(doc['content'])
+                )
+                if label_text in full_text:
+                    docs_i.append(BLANK_DOC)
+                    scors_i.append(0)
+                else:
+                    doc_chunks = [
+                        dc[0]
+                        for dc in self.pick_chunk(
+                            sq, title, full_text, url
+                        )  # type: ignore
+                    ]
+                    for splt_id, splt_content in enumerate(doc_chunks):
+                        docs_i.append(
+                            Document(
+                                docid=url, text=splt_content, title=f'{title}_{splt_id}'
+                            )
+                        )
+                        scors_i.append(self.rank_score(i))  # type: ignore
+            max_n_docs = max(max_n_docs, len(docs_i))
+            top_docs.append(docs_i)
+            top_doc_scores.append(scors_i)
+        # Pad with empty docs
+        for i in range(len(top_docs)):
+            n_empty = max_n_docs - len(top_docs[i])
+            if n_empty:
+                top_docs[i] = top_docs[i] + [BLANK_DOC] * n_empty
+                top_doc_scores[i] = top_doc_scores[i] + [0] * n_empty
+        self.top_docs = top_docs
+        self.search_queries = search_queries
+        return top_docs, torch.Tensor(top_doc_scores).to(query.device)
+
+
+class FilterDocsForLabelSearchEngineRetrieverCombo(
+    FilterDocsForLabelSearchEngineRetrieverMixin,
+    ComboFidSearchQuerySearchEngineRetriever,
+):
+    pass
+
+
+class FilterDocsForLabelSearchEngineRetrieverBase(
+    FilterDocsForLabelSearchEngineRetrieverMixin, SearchQuerySearchEngineRetriever
+):
+    pass
+
+
+#######################################
+# Special Modules/Agents for GPT2 FiD #
+#######################################
+
+
+class IdentityLayer(torch.nn.Module):
```

Comment:
I remember Spencer had added an IdentityLayer to ParlAI recently. Is that possible to use that one instead of the custom one here?